ratis-server/src/main/java/org/apache/ratis/server/impl/DataStreamServerImpl.java

```diff
@@ -41,15 +41,15 @@ public DataStreamServerImpl(RaftPeer server, StateMachine stateMachine,
       RaftProperties properties, Parameters parameters){
     final SupportedDataStreamType type = RaftConfigKeys.DataStream.type(properties, LOG::info);
 
-    this.serverRpc = DataStreamServerFactory.cast(type.newFactory(parameters))
+    this.serverRpc = DataStreamServerFactory.cast(type.newServerFactory(parameters))
         .newDataStreamServerRpc(server, stateMachine, properties);
   }
 
   public DataStreamServerImpl(RaftServer server, StateMachine stateMachine,
       RaftProperties properties, Parameters parameters){
     final SupportedDataStreamType type = RaftConfigKeys.DataStream.type(properties, LOG::info);
 
-    this.serverRpc = DataStreamServerFactory.cast(type.newFactory(parameters))
+    this.serverRpc = DataStreamServerFactory.cast(type.newClientFactory(parameters))
```

Comment:
newClientFactory -> newServerFactory ?

Reply:
Good catch.  It should be newServerFactory.